pkg/backend/httpstate/backend.go

```diff
@@ -1594,6 +1597,142 @@ func (b *cloudBackend) UpdateStackTags(ctx context.Context,
 	return b.client.UpdateStackTags(ctx, stackID, tags)
 }
 
+func (b *cloudBackend) RunDeployment(ctx context.Context, stackRef backend.StackReference,
+	req apitype.CreateDeploymentRequest, opts display.Options) error {
+
+	stackID, err := b.getCloudStackIdentifier(stackRef)
+	if err != nil {
+		return err
+	}
+
+	resp, err := b.client.CreateDeployment(ctx, stackID, req)
+	if err != nil {
+		return err
+	}
+	id := resp.ID
+
+	fmt.Printf(opts.Color.Colorize(colors.SpecHeadline + "Preparing deployment..." + colors.Reset + "\n\n"))
+
+	if !opts.SuppressPermalink && !opts.JSONDisplay {
+		if resp.ConsoleURL != "" {
+			fmt.Printf(opts.Color.Colorize(
+				colors.SpecHeadline+"View Live: "+
+					colors.Underline+colors.BrightBlue+"%s"+colors.Reset+"\n"), resp.ConsoleURL)
+		}
+	}
+
+	token := ""
+	for {
+		logs, err := b.client.GetDeploymentLogs(ctx, stackID, id, token)
+		if err != nil {
+			return err
+		}
+
+		for _, l := range logs.Lines {
+			if l.Header != "" {
+				fmt.Printf(opts.Color.Colorize(
+					"\n" + colors.SpecHeadline + l.Header + ":" + colors.Reset + "\n"))
+
+				// If we see it's a Pulumi operation, rather than outputting the deployment logs,
+				// find the associated update and show the normal rendering of the operation's events.
+				if l.Header == "Pulumi operation" {
+					fmt.Println()
+					err = b.showDeploymentEvents(ctx, stackID, apitype.UpdateKind(req.Operation.Operation), id, opts)
+					if err != nil {
+						return err
+					}
+					return nil
+				}
+			} else {
+				fmt.Print(l.Line)
+			}
+		}
+
+		// If there are no more logs for the deployment and the deployment has finished or we're not following,
+		// then we're done.
+		if logs.NextToken == "" {
+			break
+		}
+
+		// Otherwise, update the token, sleep, and loop around.
+		if logs.NextToken == token {
+			time.Sleep(500 * time.Millisecond)
+		}
+		token = logs.NextToken
+	}
+
+	return nil
+}
+
+func (b *cloudBackend) showDeploymentEvents(ctx context.Context, stackID client.StackIdentifier,
+	kind apitype.UpdateKind, deploymentID string, opts display.Options) error {
+
+	getUpdateID := func() (string, error) {
+		var tries int
+		for tries < 10 {
```

Comment:
Why is this a while-style loop instead of a `for tries := 0; tries < 10; tries++ {` style loop?